### ISSUE TITLE
Unable to load onnx file

### ISSUE THREAD

**Issue by ttdd11** [NVIDIA/TensorRT#203]
Following the mnist example, this is the output:
```
[10/07/2019-12:06:19] [I] Building and running a GPU inference engine for Onnx MNIST
----------------------------------------------------------------
Input filename:   C:/Users/tmp.onnx
ONNX IR version:  0.0.6
Opset version:    10
Producer name:    tf2onnx
Producer version: 1.5.3
Domain:
Model version:    0
Doc string:
----------------------------------------------------------------
WARNING: ONNX model has a newer ir_version (0.0.6) than this parser was built against (0.0.3).
While parsing node number 0 [Sub]:
ERROR: ModelImporter.cpp:296 In function importModel:
[5] Assertion failed: tensors.count(input_name)
```

## Environment
TensorRT 6.0
1080ti, 2080ti, Titan RTX
Driver 430
Cuda 10.1, cudnn 7.6
Windows 10

I have verified the .onnx using `onnx.checker.check_model` which returns no error, and can run the model in python. Any help getting this sorted would be appreciated. 

From onnx-tensorrt, it may be in the parser code that is throwing this issue.
 
The node it is referring to is a subtraction operation (constant) immediately after the input node, so I don't think that is the issue. 

Does my exported onnx file need to have a fixed batch size, or can I set that in trt after the model is loaded?

This is the top of the net if helpful.

![net](https://user-images.githubusercontent.com/39801645/68432124-9d916700-0168-11ea-8f3f-ffe7446b3b19.png)

**Comment by rmccorm4** [comment]
Hi @ttdd11,

1. Can you share original TF model that you used tf2onnx on?
2. Can you share your ONNX model?
3. Can you share the exact command you ran to convert?

**Comment by ttdd11** [comment]
For 1 and 2, yes, but not on this public forum. How do you recommend I send?

For 3. 
python -m tf2onnx.convert --input c:/tf_net.pb --output tmp.onnx --inputs tower_0/Placeholder:0 --outputs BatchNorm/batchnorm/add_1:0

**Comment by rmccorm4** [comment]
You can email me the above: rmccormick@nvidia.com

**Comment by ttdd11** [comment]
Thanks, sending now.

**Comment by rmccorm4** [comment]
@ttdd11, what versions are you using for tf + tf2onnx? Your conversion command isn't working for me:
```
python -m tf2onnx.convert --input tf_net.pb     --output tmp.onnx     --inputs tower_0/Placeholder:0     --outputs BatchNorm/batchnorm/add_1:0
```
Traceback
```
Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.6/dist-packages/tf2onnx/convert.py", line 161, in <module>
    main()
  File "/usr/local/lib/python3.6/dist-packages/tf2onnx/convert.py", line 116, in main
    graph_def, inputs, outputs = loader.from_graphdef(args.graphdef, args.inputs, args.outputs)
  File "/usr/local/lib/python3.6/dist-packages/tf2onnx/loader.py", line 64, in from_graphdef
    frozen_graph = freeze_session(sess, output_names=output_names)
  File "/usr/local/lib/python3.6/dist-packages/tf2onnx/loader.py", line 37, in freeze_session
    output_names, freeze_var_names)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/util/deprecation.py", line 324, in new_func
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/graph_util_impl.py", line 270, in convert_variables_to_constants
    inference_graph = extract_sub_graph(input_graph_def, output_node_names)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/util/deprecation.py", line 324, in new_func
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/graph_util_impl.py", line 182, in extract_sub_graph
    _assert_nodes_are_present(name_to_node, dest_nodes)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/graph_util_impl.py", line 137, in _assert_nodes_are_present
    assert d in name_to_node, "%s is not in graph" % d
AssertionError: BatchNorm/batchnorm/add_1 is not in graph
```
Versions - Same error for both tf1.14 and tf 1.13
```
root@6dbdfed76678:/mnt# pip freeze | grep -e tensorflow -e tf
tensorflow==1.13.2
tensorflow-estimator==1.13.0
tf2onnx==1.5.3

root@6dbdfed76678:/mnt# pip freeze | grep -e tensorflow -e tf
tensorflow==1.14.0
tensorflow-estimator==1.14.0
tf2onnx==1.5.3
```

**Comment by rmccorm4** [comment]
And for some reason ONNX parser via python API can't get any layers after parsing your model. I've never really had any issues like this. 
```
root@6dbdfed76678:/mnt# python onnx_to_trt.py --onnx=tmp.onnx 
2019-11-07 23:08:31 - __main__ - INFO - Parsing model file...
2019-11-07 23:08:31 - __main__ - ERROR - Parsed network has 0 layers. Exiting.
```
Works fine on more standard models like resnet50
```
root@6dbdfed76678:/mnt# python onnx_to_trt.py --onnx=resnet50/model.onnx 
2019-11-07 23:16:19 - __main__ - INFO - Parsing model file...
2019-11-07 23:16:19 - __main__ - INFO - Building Engine...
2019-11-07 23:16:35 - __main__ - INFO - Writing engine to model.engine
```
However, trtexec can repro your issue
```
Input filename:   tmp.onnx
ONNX IR version:  0.0.6
Opset version:    10
Producer name:    tf2onnx
Producer version: 1.5.3
Domain:           
Model version:    0
Doc string:       
----------------------------------------------------------------
WARNING: ONNX model has a newer ir_version (0.0.6) than this parser was built against (0.0.3).
While parsing node number 0 [Sub]:
ERROR: ModelImporter.cpp:296 In function importModel:
[5] Assertion failed: tensors.count(input_name)
[10/07/2019-23:14:18] [E] Failed to parse onnx file
[10/07/2019-23:14:18] [E] Parsing model failed
[10/07/2019-23:14:18] [E] Engine could not be created
&&&& FAILED TensorRT.trtexec # trtexec --onnx=tmp.onnx
```

**Comment by rmccorm4** [comment]
Hi @ttdd11,

Turns out that it's expected that this model can't be parsed.

TensorRT only supports 2D zero padding on the spatial dimensions (i.e. HW dimensions of a NCHW tensor). Padding values of [0, 1, 1, 0, 0, 1, 1, 0] implies padding on the first and second dimensions of a tensor (CH dimensions of a NCHW tensor), which is something TensorRT does not support - so this parser assertion is valid.

**Comment by ttdd11** [comment]
Thanks for the help. Is the error occurring on the first node [Sub]? If not what node are you referring to?

**Comment by ttdd11** [comment]
The error I was getting was probably related to the parser and the opset (I used 10).

To get this to parse using tensorrt, I set opset to 7, remove the layers that were not supported (including make the batch size not dynamic) and use -fold_ops parameter.

The model doesn't produce the correct results, but I think this is the onnx parser used in tensorrt and related to the upsample node (probably the opset again). Will open another issue specific to this.
